src/gt4py/next/common.py

```diff
@@ -811,13 +813,56 @@ def remapping(cls) -> ConnectivityKind:
         return cls.ALTER_DIMS | cls.ALTER_STRUCT
 
 
+@dataclasses.dataclass(frozen=True)
+class ConnectivityType:  # TODO(havogt): would better live in type_specifications but would have to solve a circular import
```

Comment:
I agree with the ToDo and I'd have a proposal on how to do it but since this PR is quite large and touches many parts, I prefer to get this merged now and implement the ToDo in a following PR.